health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/hooks/useFetchFromMicroplan.js

```
@@ -0,0 +1,172 @@
+
+    if (!campaignData?.boundaries || campaignData?.boundaries?.length==0) {
+        throw new Error("Boundary data is missing or incomplete.");
+      }
+    // Create a new campaign
+    const newCampaignResponse = await createCampaign(campaignData, tenantId, planConfigId);
+
+    if (!newCampaignResponse) {
+      throw new Error("Failed to create campaign.");
+    }
+
+    const newCampaignSearchResponse = newCampaignResponse&&newCampaignResponse?.id&& await  searchCampaign(newCampaignResponse?.id,tenantId);
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)